cmd/gonomics/gonomics.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/exception"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+)
+
+func usage() {
+	fmt.Print(
+		"gonomics - A collection of builtin tools that use the gonomics core library.\n\n" +
+			"Usage: gonomics <command> [options]\n\n")
+	printCmdList()
+}
+
+func getBin() (path string, binExists map[string]bool) {
+	path = os.Getenv("GOBIN")
+
+	if path == "" { // if GOBIN is not set
+		//fmt.Println("WARNING: GOBIN is not set, defaulting to $GOPATH/bin")
+		path = os.Getenv("GOPATH") + "/bin" // default to $GOPATH/bin
+	}
+
+	dir, err := ioutil.ReadDir(path)
+	if err != nil {
+		log.Printf("ERROR: could not access %s\n", path)
+		log.Fatal(err)
+	}
+
+	binExists = make(map[string]bool)
+
+	for _, file := range dir {
+		binExists[file.Name()] = true
+	}
+
+	if !binExists["gonomics"] { // check if gonomics is installed by checking for this binary
+		log.Fatalf("ERROR: gonomics binary were not found in %s\n "+
+			"Run 'go install ./...' in gonomics directory to compile binary", path)
+	}
+
+	return path, binExists
+}
+
+func getGonomicsCmds() map[string]bool {
+	expectedPath := os.Getenv("GOPATH") + "/src/github.com/vertgenlab/gonomics/cmd"
+	cmds, err := ioutil.ReadDir(expectedPath)
+	if err != nil {
+		log.Printf("ERROR: could not find gonomics cmd folder in expected path: %s\n", expectedPath)
+		log.Fatal(err)
+	}
+
+	funcNames := make(map[string]bool)
+
+	for _, x := range cmds {
+		funcNames[x.Name()] = true
+	}
+
+	if !funcNames["gonomics"] { // check that the directory for this cmd was found
+		log.Fatalf("ERROR: gonomics cmd folder was not found in %s\n", expectedPath)
+	}
+
+	return funcNames
+}
+
+func printCmdList() {
+	binPath, _ := getBin()
+	cmdMap := getGonomicsCmds()
+
+	cmds := make([]string, 0, len(cmdMap))
+	for key := range cmdMap {
+		cmds = append(cmds, key)
+	}
+	sort.Slice(cmds, func(i, j int) bool { return cmds[i] < cmds[j] })
+
+	var endFirstLineIdx int
+	var rawOutput []byte
+
+	fmt.Println("Commands:")
+	for _, cmdName := range cmds {
+		cmd := exec.Command(binPath + "/" + cmdName)
```

Comment:
Is this slow?  Probably not, but another option would be to parse the source (might also be slow).  Another option for the future is that we could have a github action write this program... then it would not be live, so maybe it is better to have this main gonomics program reflect the current state of all the programs and not what their state was at the last check-in/merge.